File: mobile-customer-app/lib/models/saved_address_model.dart
class SavedAddressModel {
  final String id;
  final String label;
  final String address;
  final double? lat;
  final double? lng;

  SavedAddressModel({required this.id, required this.label, required this.address, this.lat, this.lng});

  factory SavedAddressModel.fromJson(Map<String, dynamic> json) {
    return SavedAddressModel(
      id: json['_id'] as String,
      label: json['label'] as String,
      address: json['address'] as String,
      lat: (json['lat'] as num?)?.toDouble(),
      lng: (json['lng'] as num?)?.toDouble(),
    );
  }
}

File: backend/src/models/savedAddress.model.js
const mongoose = require('mongoose');

// A customer's saved pickup/drop addresses (e.g. "Home", "Office") for
// quick reuse during booking, instead of re-typing/re-searching every time.
const savedAddressSchema = new mongoose.Schema(
  {
    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    label: { type: String, required: true },
    address: { type: String, required: true },
    lat: Number,
    lng: Number,
  },
  { timestamps: true }
);

savedAddressSchema.index({ userId: 1 });

module.exports = mongoose.model('SavedAddress', savedAddressSchema);
